minio/minio.go

```diff
@@ -19,14 +19,35 @@ import (
 )
 
 type MinioI interface {
-	UploadFile(ctx context.Context, filePath string, fileContent any, fileMimeType string) (url string, objectInfo *miniogo.ObjectInfo, err error)
-	UploadFileBytes(ctx context.Context, logger *zap.Logger, filePath string, fileBytes []byte, fileMimeType string) (url string, objectInfo *miniogo.ObjectInfo, err error)
+	UploadFile(ctx context.Context, logger *zap.Logger, param *UploadFileParam) (url string, objectInfo *miniogo.ObjectInfo, err error)
+	UploadFileBytes(ctx context.Context, logger *zap.Logger, param *UploadFileBytesParam) (url string, objectInfo *miniogo.ObjectInfo, err error)
 	DeleteFile(ctx context.Context, logger *zap.Logger, filePath string) (err error)
 	GetFile(ctx context.Context, logger *zap.Logger, filePath string) ([]byte, error)
 	GetFilesByPaths(ctx context.Context, logger *zap.Logger, filePaths []string) ([]FileContent, error)
 }
 
-const Location = "us-east-1"
+type ExpiryGroup string
```

Comment:
❓ What's the benefit of creating a type over using `string` in this case?

Reply:
For usage similar to enum in golang, to specify that the input parameter should be 1 of the const defined below

Reply:
But this isn't exactly an enum but a type alias, you could do `UploadFileParam{ExpiryGroup: "garbage"}` with both types (`ExpiryGroup` and `string). From the client point of view, the valuable part in your code is the constants you added, which indicate the possible values and prevent hard-coding the values in the client code.

If you want to add an extra layer of robustness and discourage arbitrary values, there are several strategies, like keeping the value unexported and only setting it through public methods:

```go
// ExpiryGroup should be documented, as it's exported (TODO).
// Client code should use the public creators, e.g.
// `UploadFileParam{ExpiryGroup: EnterprisePlanExpiry()}`
type ExpiryGroup struct {
	str string
}

// String returns the string representation of an ExpiryGroup.
func (eg ExpiryGroup) String() string {
	return eg.str
}

// FreePlanExpiry ...
func FreePlanExpiry() ExpiryGroup { return ExpiryGroup{str: "free-plan-expiry"} }

// ProPlanExpiry ...
func ProPlanExpiry() ExpiryGroup { return ExpiryGroup{str: "pro-plan-expiry"} }

// TeamPlanExpiry ...
func TeamPlanExpiry() ExpiryGroup { return ExpiryGroup{str: "team-plan-expiry"} }

// EnterprisePlanExpiry ...
func EnterprisePlanExpiry() ExpiryGroup { return ExpiryGroup{str: "enterprise-plan-expiry"} }
```

Furthermore, you can add all the functionality you're using (tags & expiration duration) in the same struct. The tag and expiration values seem to be used internally only (correct me if I'm wromg), so you don't need to expose them.

```go
// ExpiryGroup should be documented, as it's exported (TODO).
// Client code should use the public creators, e.g.
// `UploadFileParam{ExpiryGroup: EnterprisePlanExpiry()}`
type ExpiryGroup struct {
	// these fields are only accessed internally, so we don't need public
	// methods for them.
	tag        string
	expiration time.Duration
}

// FreePlanExpiry ...
func FreePlanExpiry() ExpiryGroup {
	return ExpiryGroup{
		tag:      "free-plan-expiry",
		duration: time.Hour * 24 * 3, // 3 days
	}
}
```

This is just one of the possible strategies, my point here is that if the only method that `ExpiryGroup` implements is `String()`, from the client point of view this type is a `string` (even worse because it forces the client to do some casting but it doesn't prevent the client from misusing the type and enter invalid values).